example/android/build.gradle

```diff
@@ -71,30 +71,31 @@ gradle.ext.component_type = gradle.ext.component_type_default
 
 dependencies {
 
-    if (gradle.ext.example_as_root) {
-        // if the example is the root we use a maven dependency instead of the project
-        implementation "com.splendo.kaluga:architecture-compose:$gradle.ext.library_version"
-        implementation "com.splendo.kaluga:resources-compose:$gradle.ext.library_version"
-        implementation project(':shared')
+
+    if (gradle.ext.example_embedding_method == "as-root") {
```

Comment:
I think it's inverted statement, if 'as-root' logic for maven?

Reply:
no, `as-root` is not for maven.  It is a replacement for boolean`exampleAsRoot`. Since we have two non-maven options: `as-root` and `composite`, I decided to change it to a string instead of adding a new boolean.
Possible options : `as-root`, `maven` and `coomposite` (see in readme)

Reply:
`example_as_root == true` -> 'as-root'
`example_as_root == false` -> 'maven | composite'

Reply:
I see :+1:

Reply:
Nice work Dennis!
I haven't worked on Kaluga for a while, but commenting anyway with insights about what I remember about this task.
Consider this comment not blocking:
- do we even need `example_as_root (true)` scenario anymore?

When I started working on it, my main goal was to remove (not replace) this logic entirely: if you want to build the example you should just use the gradle tasks from the example project while everything from kaluga is build as composite. On the other hand, the main kaluga gradle should not be aware of the example project, and should build everything as in 'maven' scenario.

Does it make sense to you?
Do you think it would be achievable this way? I think it should be possible with a bit of restructuring, but cannot remember the details right now.

Reply:
@thoutbeckers yep. I don't think we really need `as-root` now but from our discussion I understood that example should support different options so I decided to keep as-root as well. Don't think it does any harm especially because we have `composite` by default.

Reply:
@posytive we discussed it with Tijl and agreed that it should support at least composite and maven

Reply:
the options refer to being able to set different maven repos and switching off composite builds. I think the former is preserved, and the latter is also supported, so as-root can be removed.

Reply:
the ticket says 
- maven local (existing before the ticket)
- remote maven support (implemented since the ticket was opened as #363)
- composite

it mentions the current "hacky way of including kaluga modules" (aka "as-root"), so that implies removing it.

Reply:
Don't see it in the related ticked but that's fine. I'll remove as-root. Basically keeping it as an extra option was more challenging than if I would just remove it

Reply:
*removed 'as-root'